test/vfd_swmr_bigset_writer.c

```diff
@@ -494,52 +703,133 @@ state_init(state_t *s, int argc, char **argv)
+    }
 
-    if (H5Pset_chunk_cache(s->dapl, 0, 0, H5D_CHUNK_CACHE_W0_DEFAULT) < 0)
-        errx(EXIT_FAILURE, "H5Pset_chunk_cache failed");
+    if (s->vds != vds_off && H5Pset_virtual_view(s->dapl, H5D_VDS_FIRST_MISSING) < 0) {
+        fprintf(stderr, "H5Pset_virtual_view failed\n");
+        TEST_ERROR;
+    }
+
+    return true;
+
+error:
+    H5E_BEGIN_TRY
+    {
+        H5Pclose(s->quadrant_dcpl);
+        H5Sclose(ul->space);
+        H5Sclose(ur->space);
+        H5Sclose(bl->space);
+        H5Sclose(br->space);
+        H5Sclose(ul->src_space);
+        H5Sclose(ur->src_space);
+        H5Sclose(bl->src_space);
+        H5Sclose(br->src_space);
+        H5Sclose(src->space);
+        H5Sclose(s->one_by_one_sid);
+        H5Sclose(s->memspace);
+    }
+    H5E_END_TRY;
+
+    HDfree(tfile);
 
-    if (s->vds != vds_off && H5Pset_virtual_view(s->dapl, H5D_VDS_FIRST_MISSING) < 0)
-        errx(EXIT_FAILURE, "H5Pset_virtual_view failed");
+    HDfree(s->dataset);
+
+    HDfree(s->sources);
+
+    return false;
 }
 
-static void
+static bool
 state_destroy(state_t *s)
 {
-    size_t i;
-
-    if (H5Pclose(s->dapl) < 0)
-        errx(EXIT_FAILURE, "H5Pclose(fapl)");
+    size_t          i;
+    struct timespec start_time, end_time;
 
-    s->dapl = badhid;
+    if (H5Pclose(s->dapl) < 0) {
+        fprintf(stderr, "H5Pclose failed\n");
+        TEST_ERROR;
+    }
 
     if (s->vds != vds_off) {
         quadrant_t *const ul = &s->quadrants.ul, *const ur = &s->quadrants.ur, *const bl = &s->quadrants.bl,
                           *const br = &s->quadrants.br;
 
         if (H5Sclose(ul->src_space) < 0 || H5Sclose(ur->src_space) < 0 || H5Sclose(bl->src_space) < 0 ||
-            H5Sclose(br->src_space) < 0)
-            errx(EXIT_FAILURE, "H5Sclose failed");
+            H5Sclose(br->src_space) < 0) {
+            fprintf(stderr, "H5Sclose failed\n");
+            TEST_ERROR;
+        }
+
+        if (H5Sclose(ul->space) < 0 || H5Sclose(ur->space) < 0 || H5Sclose(bl->space) < 0 ||
+            H5Sclose(br->space) < 0) {
+            fprintf(stderr, "H5Sclose failed\n");
+            TEST_ERROR;
+        }
+
+        if (H5Pclose(s->quadrant_dcpl) < 0) {
+            fprintf(stderr, "H5Pclose failed\n");
+            TEST_ERROR;
+        }
+    }
+
+    if (H5Sclose(s->one_by_one_sid) < 0) {
+        fprintf(stderr, "H5Sclose failed\n");
+        TEST_ERROR;
+    }
 
-        ul->src_space = ur->src_space = bl->src_space = br->src_space = badhid;
+    if (H5Sclose(s->memspace) < 0) {
+        fprintf(stderr, "H5Sclose failed\n");
+        TEST_ERROR;
+    }
 
-        if (H5Pclose(s->quadrant_dcpl) < 0)
-            errx(EXIT_FAILURE, "H5Pclose(dcpl)");
+    /* The writer ends the tick before closing the file to make the data visible to the reader */
+    if (s->writer && s->use_vfd_swmr) {
+        unsigned long j;
 
-        s->quadrant_dcpl = badhid;
+        if (s->vds != vds_multi)
+            H5Fvfd_swmr_end_tick(s->file[0]);
```

Comment:
Why not checking the error message for this call and the call three lines below?

Reply:
Fixed.